test_datasets/10x_pbmc_hto_GSE108313/download.R

```diff
@@ -31,31 +43,20 @@ pbmc_umi = pbmc_umi[,joint_barcodes]
 pbmc_hto = pbmc_hto[,joint_barcodes]
 pbmc_counts = rbind(pbmc_umi, pbmc_hto)
 
-# Write to 3 CellRanger output files
-DropletUtils::write10xCounts(path="./counts", x=pbmc_counts)
-
-# Add HTO to genes.tsv
-genes = read.delim("./counts/genes.tsv", header=FALSE)
-genes = cbind(genes, "Gene Expression")
-colnames(genes) = c("Ensembl", "GeneSymbol", "Type")
-genes$Type = as.character(genes$Type)
-genes$Type[grep("^hto", genes$GeneSymbol, perl=TRUE)] = "Antibody Capture"
-
-# Add Ensembl IDs to features.tsv
-genes$Ensembl = c(genes_ensembl[genes$GeneSymbol], hto_names)
+feat = data.frame(V1=genes_ensembl[rownames(pbmc_umi)], V2=rownames(pbmc_umi), V3="Gene Expression")
+feat = rbind(feat, data.frame(V1=rownames(pbmc_hto), V2=rownames(pbmc_hto), V3="Antibody Capture"))
 
-# Write genes to features.tsv
-write.table(genes, file="counts/features.tsv", sep="\t", row.names=FALSE, col.names=FALSE, quote=FALSE)
+# write matrix.mtx.gz, barcodes.tsv.gz and features.tsv.gz
+mh = file.path("counts","matrix.mtx")
+Matrix::writeMM(pbmc_counts,file=mh)
+R.utils::gzip(mh,overwrite=T)
 
-# Gzip all three files
-system("gzip counts/barcodes.tsv")
-system("gzip counts/features.tsv")
-system("gzip counts/matrix.mtx")
+bh = gzfile(file.path("counts", "barcodes.tsv.gz"), open="wb")
+write(colnames(pbmc_counts), file=bh)
+close(bh)
 
-# Remove genes.tsv
-# This is necessary, so Seurat recognises the CellRanger version
-system("rm counts/genes.tsv")
```

Comment:
I vaguely remember that removing `genes.tsv` was necessary. In your case, is it not written in the first place?